transforms/helpers/ast.ts

```diff
@@ -69,91 +71,93 @@ export function isNode<T extends ASTNode['type'] = ASTNode['type']>(
 
 export interface EOExtendExpression extends CallExpression {
   callee: EOExtendExpressionCallee;
+  arguments: Array<EOExpression | EOMixin>;
 }
 
 export function isEOExtendExpression(u: unknown): u is EOExtendExpression {
-  return isNode(u, 'CallExpression') && isEOExtendExpressionCallee(u.callee);
+  return (
+    isNode(u, 'CallExpression') &&
+    isEOExtendExpressionCallee(u.callee) &&
+    u.arguments.every(
+      (argument) => isEOExpression(argument) || isEOMixin(argument)
+    )
+  );
 }
 
-type EOExtendExpressionCallee = CalleeWithObjectAndProperty & {
-  object: EOExtendExpressionCalleeObject;
+interface EOExtendExpressionCallee extends MemberExpression {
+  object: Identifier;
   property: EOExtendExpressionCalleeProperty;
-};
+}
 
 function isEOExtendExpressionCallee(u: unknown): u is EOExtendExpressionCallee {
   return (
-    isNode(u) &&
-    'object' in u &&
-    isEOExtendExpressionCalleeObject(u.object) &&
-    'property' in u &&
+    (isNode(u, 'MemberExpression') || isNode(u, 'OptionalMemberExpression')) &&
+    isNode(u.object, 'Identifier') &&
     isEOExtendExpressionCalleeProperty(u.property)
   );
 }
 
-type CalleeWithObjectAndProperty = Extract<
-  CallExpression['callee'],
-  { object: unknown; property: unknown }
->;
-
-type NonNullCalleeObject = Exclude<CalleeWithObjectAndProperty['object'], null>;
-
-type EOExtendExpressionCalleeObject = Extract<
-  NonNullCalleeObject,
-  { name: string }
->;
-
-function isEOExtendExpressionCalleeObject(
-  u: unknown
-): u is EOExtendExpressionCalleeObject {
-  return (
-    isNode(u) &&
-    'name' in u &&
-    typeof u.name === 'string' &&
-    startsWithUpperCaseLetter(u.name)
-  );
-}
-
-type CalleePropertyWithStringName = Extract<
-  CalleeWithObjectAndProperty['property'],
-  { name?: string }
->;
-
-type EOExtendExpressionCalleeProperty = CalleePropertyWithStringName & {
+interface EOExtendExpressionCalleeProperty extends Identifier {
   name: 'extend';
-};
+}
 
 function isEOExtendExpressionCalleeProperty(
   u: unknown
 ): u is EOExtendExpressionCalleeProperty {
-  return isNode(u) && 'name' in u && u.name === 'extend';
+  return isNode(u, 'Identifier') && u.name === 'extend';
 }
 
-type EOExtendExpressionArgs = EOExtendExpression['arguments'];
-
-type EOExtendExpressionArg = EOExtendExpressionArgs[number];
-
 export interface EOExpression extends ObjectExpression {
-  properties: EOProperty[];
+  properties: Array<EOProperty | EOMethod>;
 }
 
 export function isEOExpression(u: unknown): u is EOExpression {
-  return isNode(u, 'ObjectExpression') && u.properties.every(isEOProperty);
+  return (
+    isNode(u, 'ObjectExpression') &&
+    u.properties.every(
+      (property) => isEOProperty(property) || isEOMethod(property)
+    )
+  );
 }
 
-// NOTE: This can probably be narrowed further if necessary.
-export type EOMixin = Exclude<EOExtendExpressionArg, ObjectExpression>;
+export interface EOMixin extends Identifier {}
```

Comment:
Since this isn't a nominal type, maybe just:

```suggestion
export type EOMixin = Identifier;
```